# AirBnB Data Analysis Project

This is a capstone project for the Code:You Data Analysis course. This project 
will analyze Airbnb data to uncover useful correlations/trends and guide 
decision making in investing in STR properties.

## Project Goal

**Determine the optimal location and house size for a short term rental investment 
property in Seattle.**

The end result of this project is to classify properties in terms of STR 
Performance vs. Purchase Price:
- High STR Performance, Low Purchase Price
- High STR Performance, High Purchase Price
- Low STR Performance, Low Purchase Price
- Low STR Perormance, High Purchase Price

In this data discovery notebook, I will:
1. Research key metrics for Short Term Rentals.
1. Research the Seattle housing market.
1. Determine data needed to support.
1. Review available data.
1. Identify the type of analysis that can be done and list the questions that 
can be answered.
1. List the cleaning steps that will be needed.


## 1. Short Term Rental Metrics Research

A quick google search gives us some of the key metrics used to evaluate the 
financial performance of a short term rental property. NOI and RPP appear to be
most related to our goal of finding a good investment property.

> **Net operating income (NOI)** is another key Airbnb metric for Airbnb hosts to evaluate the profitability of their listing. NOI measures the total revenue generated by a property from all sources and subtracts operating expenses, including management, maintenance, and cleaning fees. Hosts can use NOI to accurately determine their net profit and make informed decisions about pricing strategy, cost-saving measures, or investing in property improvements. By focusing on maximizing their NOI, hosts can drive profitability on the Airbnb platform and make sure they achieve sustainable success. Therefore, tracking NOI is an essential factor for any host looking to optimize their financial performance and grow their vacation rental business.

> **Revenue per property (RPP)** calculates the total earnings generated by a property within a defined timeframe. Monitoring RPP allows hosts to discern which listings are the highest revenue generators, enabling them to fine-tune pricing and marketing strategies accordingly. Furthermore, RPP aids hosts in assessing the financial feasibility of investing in new properties or enhancing existing ones. By optimizing RPP, hosts can enhance operational efficiency on Airbnb and thrive in the fiercely competitive vacation rental market. Tracking RPP becomes a pivotal aspect for hosts seeking to maximize financial performance and expand their business endeavors.

Sources: Short Term Rental KPIs

- [Key STR Metrics](https://hostify.com/blog/key-airbnb-performance-metrics-hosts-should-track)



## 2. Seattle Real Estate Market Research 

In order to determine the best location and size of house to buy for a short
term rental, we will need a way to compare the relative price of different
properties. A quick google search yielded house prices for the Seattle Market 
broken down by neighborhood and number of bedrooms.


House Sale Price by Neighborhood

| Nieghborhood | MedianPrice |
| ------------ | ----------- |
| Auburn | $407,634 |
| Ballard/Greenlake | $779,080 |
| Bellevue–East of 405 | $894,374 |
| Bellevue–West of 405 | $2,084,935 |
| Belltown/Downtown | $849,400 |
| Black Diamond/Maple Valley | $516,151 |
| Burien/Normandy Park | $500,085 |
| Central Seattle | $932,492 |
| Dash Point/Federal Way | $404,595 |
| Des Moines/Redondo | $387,942 |
| East of Lake Sammamish | $877,393 |
| East Side–South of I-90 | $940,779 |
| Enumclaw | $473,865 |
| Jovita/West Hill | $459,289 |
| Juanita/Woodinville | $775,985 |
| Kent | $433,721 |
| Kirkland/Bridle Trails | $1,183,655 |
| Lake Forest Park | $647,703 |
| Mercer Island | $1,807,002 |
| North Seattle | $854,153 |
| Queen Anne/Magnolia | $969,366 |
| Redmond/Carnation | $872,609 |
| Renton-Benson Hill | $495,858 |
| Renton-Highlands | $596,416 |
| Richmond Beach/Shoreline | $680,157 |
| Skyway Area | $473,430 |
| SODO/Beacon Hill | $610,983 |
| Southeast Seattle | $757,231 |
| Vashon Island | $676,840 |
| West Seattle | $679,560 |



House Sale Price by Number of Bedrooms

| Bedrooms | AverageSize | MedianSalePrice | PriceDifference | PricePerSqFt | NumberOfSales |
| -------- | ----------- | --------------- | --------------- | ------------ | ------------- |
| 1 | | $678,667 | -$221,333 | | |
| 2 | 1321 | $900,000 | -$212,000 | 681 | 77 |
| 3 | 1906 | $1,112,000 | $212,000 | 583 | 193 
| 4	| 2420 | $1,402,500 | $290,000 | 579 | 110 |
| 5 or more | 2930 | $1,565,000 | $162,000 | 534 | 51 |

**Sources** 

Seattle Real Estate Market Data

- [Home Price by Neighborhood](https://www.seattlemet.com/home-and-real-estate/how-much-do-homes-cost-in-seattle-area-neighborhoods-real-estate)
- [Home Price by Number of Bedrooms](https://www.myseattlehomesearch.com/blog/seattle-house-prices-versus-the-number-of-bedrooms/#:~:text=Not%20shown%20in%20the%20data,just%2012%25%20of%20the%20market.)

There are more three bedroom houses sold than any other size, so we can use 
that as the starting point, and adjust for houses with fewer or more bedrooms.

| Bedrooms | BedroomAdjustment |
| -------- | ----------------- |
| 1 | -39% |
| 2 | -24% |
| 3 | 0 |
| 4	| 26% |
| 5 or more | 41% |



## 3. Data Needed To Support Topic

Housing Market
- Sales Price 
- number of bedrooms
- location

AirBnB Data
- number / percent of booked days 
- number of bedrooms
- location
- nightly rate

Other (Needed for NOI calculations)
- mortgage rate - we can use 7.5% and adjust if needed
- STR management fee - we can use 10%

## 4. Data Discovery - Kaggle AirBnB Data

The next step is to see what data is available in the Kaggle data set.

Sources:
- [Kaggle Seattle Listing Data Set](https://www.kaggle.com/datasets/airbnb/seattle)


There are three primary data sets available from the Kaggle data.
- AirBnB Listings
- AirBnB Reviews
- AirBnB Calendar

### Listings

The listings data set contains AirBnB listings from Seattle, Washington. To 
start off the analysis we will load and preview the listings data.

In [39]:
import pandas as pd

listings = pd.read_excel("../data/raw/Tableau Full Project.xlsx", sheet_name=0)
listings

id                            listing_url       scrape_id  \
0       241032    https://www.airbnb.com/rooms/241032  20160104002432   
1       953595    https://www.airbnb.com/rooms/953595  20160104002432   
2      3308979   https://www.airbnb.com/rooms/3308979  20160104002432   
3      7421966   https://www.airbnb.com/rooms/7421966  20160104002432   
4       278830    https://www.airbnb.com/rooms/278830  20160104002432   
...        ...                                    ...             ...   
3813   8101950   https://www.airbnb.com/rooms/8101950  20160104002432   
3814   8902327   https://www.airbnb.com/rooms/8902327  20160104002432   
3815  10267360  https://www.airbnb.com/rooms/10267360  20160104002432   
3816   9604740   https://www.airbnb.com/rooms/9604740  20160104002432   
3817  10208623  https://www.airbnb.com/rooms/10208623  20160104002432   

     last_scraped                                 name  \
0      2016-01-04         Stylish Queen Anne Apartment   
1      2016-01-04   Bright & Airy Queen Anne Apartment   
2      2016-01-04  New Modern House-Amazing water view   
3      2016-01-04                   Queen Anne Chateau   
4      2016-01-04       Charming craftsman 3 bdm house   
...           ...                                  ...   
3813   2016-01-04   3BR Mountain View House in Seattle   
3814   2016-01-04    Portage Bay View!-One Bedroom Apt   
3815   2016-01-04    Private apartment view of Lake WA   
3816   2016-01-04    Amazing View with Modern Comfort!   
3817   2016-01-04            Large Lakefront Apartment   

                                                summary  \
0                                                   NaN   
1     Chemically sensitive? We've removed the irrita...   
2     New modern house built in 2013.  Spectacular s...   
3     A charming apartment that sits atop Queen Anne...   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...   
3814  800 square foot 1 bedroom basement apartment w...   
3815  Very comfortable lower unit. Quiet, charming m...   
3816  Cozy studio condo in the heart on Madison Park...   
3817  All hardwood floors, fireplace, 65" TV with Xb...   

                                                  space  \
0     Make your self at home in this charming one-be...   
1     Beautiful, hypoallergenic apartment in an extr...   
2     Our house is modern, light and fresh with a wa...   
3                                                   NaN   
4     Cozy family craftman house in beautiful neighb...   
...                                                 ...   
3813  Our 3BR/2BA house bright, stylish, and wheelch...   
3814  This space has a great view of Portage Bay wit...   
3815                                                NaN   
3816  Fully furnished unit to accommodate most needs...   
3817                                                NaN   

                                            description experiences_offered  \
0     Make your self at home in this charming one-be...                none   
1     Chemically sensitive? We've removed the irrita...                none   
2     New modern house built in 2013.  Spectacular s...                none   
3     A charming apartment that sits atop Queen Anne...                none   
4     Cozy family craftman house in beautiful neighb...                none   
...                                                 ...                 ...   
3813  Our 3BR/2BA house boasts incredible views of t...                none   
3814  800 square foot 1 bedroom basement apartment w...                none   
3815  Very comfortable lower unit. Quiet, charming m...                none   
3816  Cozy studio condo in the heart on Madison Park...                none   
3817  All hardwood floors, fireplace, 65" TV with Xb...                none   

                                  neighborhood_overview  ...  \
0      

In [40]:
listings.shape

(3818, 92)

There are 3,818 AirBnB listings included in the data. Each listing has 92 attributes.

Next we will look at the list of attributes available.

In [41]:
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 92 columns):
 #   Column                            Non-Null Count  Dtype         
---  ------                            --------------  -----         
 0   id                                3818 non-null   int64         
 1   listing_url                       3818 non-null   object        
 2   scrape_id                         3818 non-null   int64         
 3   last_scraped                      3818 non-null   datetime64[ns]
 4   name                              3818 non-null   object        
 5   summary                           3639 non-null   object        
 6   space                             3249 non-null   object        
 7   description                       3818 non-null   object        
 8   experiences_offered               3818 non-null   object        
 9   neighborhood_overview             2786 non-null   object        
 10  notes                             2206 non-null 

**Numeric Fields**

In [42]:
listings.describe()

id     scrape_id                   last_scraped  \
count  3.818000e+03  3.818000e+03                           3818   
mean   5.550111e+06  2.016010e+13  2016-01-04 00:00:00.000000256   
min    3.335000e+03  2.016010e+13            2016-01-04 00:00:00   
25%    3.258256e+06  2.016010e+13            2016-01-04 00:00:00   
50%    6.118244e+06  2.016010e+13            2016-01-04 00:00:00   
75%    8.035127e+06  2.016010e+13            2016-01-04 00:00:00   
max    1.034016e+07  2.016010e+13            2016-01-04 00:00:00   
std    2.962660e+06  0.000000e+00                            NaN   

            host_id                     host_since  host_response_rate  \
count  3.818000e+03                           3816         3295.000000   
mean   1.578556e+07  2013-09-18 17:26:02.264151040            0.948868   
min    4.193000e+03            2008-11-10 00:00:00            0.170000   
25%    3.275204e+06            2012-08-17 00:00:00            0.980000   
50%    1.055814e+07            2013-12-12 12:00:00            1.000000   
75%    2.590309e+07            2015-01-14 00:00:00            1.000000   
max    5.320861e+07            2016-01-03 00:00:00            1.000000   
std    1.458382e+07                            NaN            0.118667   

       host_acceptance_rate  host_listings_count  host_total_listings_count  \
count           3045.000000          3816.000000                3816.000000   
mean               0.999672             7.157757                   7.157757   
min                0.000000             1.000000                   1.000000   
25%                1.000000             1.000000                   1.000000   
50%                1.000000             1.000000                   1.000000   
75%                1.000000             3.000000                   3.000000   
max                1.000000           502.000000                 502.000000   
std                0.018122            28.628149                  28.628149   

          latitude  ...  review_scores_rating  review_scores_accuracy  \
count  3818.000000  ...           3171.000000             3160.000000   
mean     47.628961  ...             94.539262                9.636392   
min      47.505088  ...             20.000000                2.000000   
25%      47.609418  ...             93.000000                9.000000   
50%      47.623601  ...             96.000000               10.000000   
75%      47.662694  ...             99.000000               10.000000   
max      47.733358  ...            100.000000               10.000000   
std       0.043052  ...              6.606083                0.698031   

       review_scores_cleanliness  review_scores_checkin  \
count                3165.000000            3160.000000   
mean                    9.556398               9.786709   
min                     3.000000               2.000000   
25%                     9.000000              10.000000   
50%                    10.000000              10.000000   
75%                    10.000000              10.000000   
max                    10.000000              10.000000   
std                     0.797274               0.595499   

       review_scores_communication  review_scores_location  \
count                  3167.000000             3163.000000   
mean                      9.809599                9.608916   
min                       2.000000                4.000000   
25%                      10.000000                9.000000   
50%                      10.000000               10.000000   
75%                      10.000000               10.000000   
max                      10.000000               10.000000   
std                       0.568211                0.629053   

       review_scores_value  license  calculated_host_listings_count  \
count          3162.000000      0.0                     3818.000000   
mean              9.452245      NaN                        2.946307   
min               2.000000      NaN                        1.00

In [43]:
listings.price.describe()

count    3818.000000
mean      127.976166
std        90.250022
min        20.000000
25%        75.000000
50%       100.000000
75%       150.000000
max      1000.000000
Name: price, dtype: float64

**Categorical Fields**

In [44]:
listings.neighbourhood_group_cleansed.unique()

array(['Queen Anne', 'Ballard', 'Other neighborhoods', 'Cascade',
       'Central Area', 'University District', 'Downtown', 'Magnolia',
       'West Seattle', 'Interbay', 'Beacon Hill', 'Rainier Valley',
       'Delridge', 'Seward Park', 'Northgate', 'Capitol Hill',
       'Lake City'], dtype=object)

In [45]:
listings.neighbourhood_group_cleansed.value_counts()

neighbourhood_group_cleansed
Other neighborhoods    794
Capitol Hill           567
Downtown               530
Central Area           369
Queen Anne             295
Ballard                230
West Seattle           203
Rainier Valley         159
University District    122
Beacon Hill            118
Cascade                 89
Northgate               80
Delridge                79
Lake City               67
Magnolia                61
Seward Park             44
Interbay                11
Name: count, dtype: int64

**Fields to Keep and Clean**

- id
- name
- street
- neighbourhood 
- neighbourhood_cleansed 
- neighbourhood_group_cleansed 
- city    
- state    
- zipcode   
- market    
- smart_location 
- country_code 
- country
- latitude
- longitude
- is_location_exact
- property_type
- room_type
- accommodates
- bathrooms
- bedrooms
- beds  
- bed_type
- amenities
- square_feet
- price
- weekly_price
- monthly_price
- security_deposit
- cleaning_fee
- guests_included
- extra_people
- minimum_nights
- maximum_nights
- calendar_updated
- has_availability
- availability_30
- availability_60
- availability_90
- availability_365
- calendar_last_scraped
- number_of_reviews
- first_review
- last_review
- review_scores_rating
- review_scores_accuracy
- review_scores_cleanliness
- review_scores_checkin
- review_scores_communication
- review_scores_location
- review_scores_value

## 4. Questions that can be answered

The purpose of this project is to analyze AirBnB data to determine the most 
attractive attributes of a short term rental investment.


- Question 1
- Question 2

## 5. Cleaning Needed

- Remove Columns
- Rename Columns
- Filter Rows
- Create Calculated Fields